akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala

```diff
@@ -287,15 +287,12 @@ private[akka] class RemoteWatcher(
    * does not exist.
    */
   def reWatch(address: Address): Unit =
-    watching.foreach {
-      case (wee: InternalActorRef, wer: InternalActorRef) ⇒
-        if (wee.path.address == address) {
-          // this re-watch will result in a RewatchRemote message to this actor
-          // must be a special message to be able to detect if an UnwatchRemote comes in
-          // before the extra RewatchRemote, then the re-watch should be ignored
-          log.debug("Re-watch [{} -> {}]", wer, wee)
-          wee.sendSystemMessage(new Rewatch(wee, wer)) // ➡➡➡ NEVER SEND THE SAME SYSTEM MESSAGE OBJECT TO TWO ACTORS ⬅⬅⬅
-        }
+    for {
+      wee ← watching.keys if wee.path.address == address
+      (watchee: InternalActorRef, watcher: InternalActorRef) = (wee, self)
```

Comment:
I'd avoid this pattern in performance sensitive code (`(a, b) = (x, y)`) - this allocates a Tuple2 each time (this may actually be the case where 2 Tuple2 allocations happen for each such line actually if I remember correctly).
A `match` inside would be allocation free and thus faster. :-)